toolchain/lowering/lowering_context.h

```diff
@@ -38,6 +38,10 @@ class LoweringContext {
       // `type` is lowered to an empty LLVM StructType.
       return llvm::StructType::get(llvm_context());
     }
+    // Function with no return type gets void type.
+    if (type_id == SemanticsTypeId::Invalid) {
+      return llvm::Type::getVoidTy(llvm_context());
+    }
```

Comment:
I added it here for consistency with how this function is called for getting the llvm::type for all the semantic types. However, this will cause some code to break if we are not assuming all invalid types should be void. In that case, it would be better to move it to BuildFunctionDeclaration. Please let me know your thoughts.

Reply:
The use of invalid for void is limited to functions. We should not assume that will exist elsewhere in the future, so BuildFunctionDeclaration seems like the right place to me.